common/src/lib.rs

```diff
@@ -349,15 +349,20 @@ pub fn program_key(id: H256) -> Vec<u8> {
 pub fn pages_prefix(program_id: H256) -> Vec<u8> {
     let mut key = Vec::new();
     key.extend(STORAGE_PROGRAM_PAGES_PREFIX);
-    program_id.encode_to(&mut key);
+    key.extend(program_id.as_fixed_bytes());
+    key.extend(b"::");
 
     key
 }
 
 fn page_key(id: H256, page: PageNumber) -> Vec<u8> {
-    let mut key = pages_prefix(id);
+    // try to avoid realloc
+    let mut key = Vec::with_capacity(STORAGE_PROGRAM_PAGES_PREFIX.len() + 38);
```

Comment:
```suggestion
    let mut key = Vec::with_capacity(STORAGE_PROGRAM_PAGES_PREFIX.len() + 38);
```

better to avoid explicitly specified vars. we can pre-calculate len upper